src/main/java/com/synopsys/arc/jenkinsci/plugins/extratoolinstallers/installers/AuthenticatedZipExtractionInstaller.java

```diff
@@ -0,0 +1,560 @@
+                 * uri + "," + usernameOrNull + "," + passwordOrNull + "," +
+                 * timestampOfLocalContents + "," + nodeName + "," +
+                 * whereToDownloadToOrNull + "," + log + ")");
+                 */
+                AuthenticatedDownloadCallable.downloadAndUnpack(uri, usernameOrNull, passwordOrNull,
+                        timestampOfLocalContents, nodeName, whereToDownloadToOrNull, log);
+            } catch (AuthenticatedDownloadCallable.HttpGetException ex) {
+                if (ex.getHttpStatusCode() != null) {
+                    final int httpStatusCode = ex.getHttpStatusCode().intValue();
+                    if (httpStatusCode == HttpStatus.SC_UNAUTHORIZED) {
+                        if (usernameOrNull == null) {
+                            return credentialProblem(checkUrl, FormValidation.error(ex,
+                                    Messages.AuthenticatedZipExtractionInstaller_credentials_required()));
+                        } else {
+                            return credentialProblem(checkUrl, FormValidation.error(ex,
+                                    Messages.AuthenticatedZipExtractionInstaller_credentials_rejected(usernameOrNull)));
+                        }
+                    }
+                    if (httpStatusCode == HttpStatus.SC_NOT_FOUND) {
+                        return credentialProblem(checkUrl, FormValidation.error(ex,
+                                Messages.AuthenticatedZipExtractionInstaller_404_http_response_from_server()));
+                    }
+                }
+                return urlProblem(checkUrl, FormValidation.error(ex,
+                        Messages.AuthenticatedZipExtractionInstaller_bad_http_response_from_server(ex.getMessage())));
+            } catch (IOException | InterruptedException ex) {
+                return urlProblem(checkUrl,
+                        FormValidation.error(ex, Messages.AuthenticatedZipExtractionInstaller_could_not_connect()));
+            }
+            return FormValidation.ok();
+        }
+
+        private static @Nonnull FormValidation urlProblem(final boolean checkUrlNotCredentials,
+                @Nonnull final FormValidation problemWithUrl) {
+            return checkUrlNotCredentials ? problemWithUrl : FormValidation.ok();
+        }
+
+        private static @Nonnull FormValidation credentialProblem(final boolean checkUrlNotCredentials,
+                @Nonnull final FormValidation problemWithCredentials) {
+            return checkUrlNotCredentials ? FormValidation.ok() : problemWithCredentials;
+        }
+
+        private static boolean hasPermissionToConfigure(@Nullable final Item item) {
+            if (item == null) {
+                return Jenkins.getInstance().hasPermission(Jenkins.ADMINISTER);
+            }
+            return item.hasPermission(Item.EXTENDED_READ) || item.hasPermission(CredentialsProvider.USE_ITEM);

```

Comment:
I doubt it ever possible for Tool Installers. They should be configurable only by admins, and it may be better to hardcode it

Reply:
OK, I've removed `item` and the associated complexity.